library/core/functions.general.php

```diff
@@ -3052,9 +3054,25 @@ function redirectTo($destination = null, $statusCode = 302, $trustedOnly = true)
         // This would cause http://evil.domain\@trusted.domain/ to be converted to http://evil.domain/@trusted.domain/
         $url = str_replace('\\', '%5c', $url);
 
-        if (Gdn::controller() !== null
-            && in_array(Gdn::controller()->deliveryType(), [DELIVERY_TYPE_ASSET, DELIVERY_TYPE_VIEW], true)
-            && Gdn::controller()->deliveryMethod() === DELIVERY_METHOD_JSON) {
+        $controller = Gdn::controller();
+
+        // Send any headers that were put together.
+        if ($controller !== null) {
+            if ($statusCode === 302) {
+                // 302 redirects are TEMPORARY and we don't want them cached by the browser.
+                $controller->setHeader('Cache-Control', CacheControlMiddleware::NO_CACHE);
+            }
+
+            // Send the headers the controller has collected so far.
+            $controller->sendHeaders();
```

Comment:
This is potentially a larger change with other unexpected side-effects. Do you think this is necessary?

Reply:
I did find this https://github.com/vanilla/vanilla/blob/2ceba39c4cf112af807c4559231118115195161a/library/core/class.controller.php#L2101-L2102

Currently I can't think necessarily think of a particular scenario in favour or against, but I'll remove it for now.

This seems very hacky to begin with so when we want something more considered we can do it in some new class.